lib/auth/v4/headerAuthCheck.js

```diff
@@ -126,8 +126,15 @@ function check(request, log, data, awsService) {
     if (isTimeSkewed) {
         return { err: errors.RequestTimeTooSkewed };
     }
-    const proxyPath = request.headers.proxy_path ?
-    decodeURIComponent(request.headers.proxy_path) : null;
+
+    let proxyPath = null;
+    if (request.headers.proxy_path) {
+        try {
+            proxyPath = decodeURIComponent(request.headers.proxy_path);
+        } catch (err) {
+            log.debug('invalid proxy_path header', { proxyPath });
```

Comment:
@rahulreddy I was not sure if I should return an error here or not to keep consistency.

Reply:
Also, if we want to return an error, I was not sure on which one should we return. :)

Reply:
I recommend logging the `err` as a debug log and returning InvalidArgument as error.

Reply:
Thanks, I addressed your comment :)